cupy/core/core.pyx

```diff
@@ -1242,12 +1242,18 @@ cdef class ndarray:
             return NotImplemented
 
     def __array_function__(self, func, types, args, kwargs):
-        if not hasattr(cupy, func.__name__):
+        import cupy as module
```

Comment:
I think simply `module = cupy` shoud work.

Reply:
It does, thanks for spotting it. Pushing a commit in a minute.

Reply:
Thanks!